src/transformers/generation_utils.py

```diff
@@ -1882,28 +1882,34 @@ def contrastive_search(
             model_inputs = self.prepare_inputs_for_generation(input_ids, **model_kwargs)
 
             # if the first step in the loop, encode all the prefix and obtain three parameters: (1) past_key_values;
-            # (2) last_hidden_states; (3) logit_for_next_step
+            # (2) last_hidden_states; (3) logit_for_next_step; (4) update model kwargs for the next step
             if model_kwargs.get("past") is None:
                 # encode the given prefix and prepare model inputs; encoder-decoder model process the prefix and save
                 # the `encoder_outputs`
-                output = self(**model_inputs, output_hidden_states=True, output_attentions=True)
+                outputs = self(
+                    **model_inputs, return_dict=True, output_hidden_states=True, output_attentions=output_attentions
+                )
```

Comment:
Two changes here:
1. makes `return_dict` explicitly true (needed for contrastive search) and `output_attentions` optional (as in the other functions)
2. `output` -> `outputs`, as in the other functions (to facilitate copy/paste)